rust/experimental/query_abstraction/ottl_plugin/src/ottl_plugin.rs

```diff
@@ -1,15 +1,864 @@
+                            BinaryLogicalExpression {
+                                left: Box::new(Expression::EnclosedExpression(Box::new(
+                                    Expression::Predicate(Predicate::ComparisonExpression(
+                                        ComparisonExpression {
+                                            left: Box::new(Expression::Identifier(Identifier {
+                                                name: "b".to_string(),
+                                            })),
+                                            comparison_operator: ComparisonOperator::Equal,
+                                            right: Box::new(Expression::Literal(Literal::Int(5))),
+                                        }
+                                    ))
+                                ))),
+                                boolean_operator: BooleanOperator::And,
+                                right: Box::new(Expression::EnclosedExpression(Box::new(
+                                    Expression::Predicate(Predicate::ComparisonExpression(
+                                        ComparisonExpression {
+                                            left: Box::new(Expression::Identifier(Identifier {
+                                                name: "c".to_string(),
+                                            })),
+                                            comparison_operator: ComparisonOperator::NotEqual,
+                                            right: Box::new(Expression::Literal(Literal::String(
+                                                "world".to_string()
+                                            ))),
+                                        }
+                                    ))
+                                ))),
+                            }
+                        ))
+                    )
+                ]
+            }
+        );
+    }
+
+    #[test]
+    fn test_process_query_filter_complex() {
+        let input = r#"filter:
+        logs:
+        log_record:
+        - 'x == 42'
+        - 'y > 10'
+        - 'not(z <= 5)'
+        - '(a == true) or (b != "test")'"#;
+        let result = OttlPlugin::process_query(input);
+        assert!(result.is_ok());
+        let query = result.unwrap();
+        assert_eq!(
+            query,
+            Query {
+                source: "logs".to_string(),
+                statements: vec![
+                    Statement::Filter(Predicate::ComparisonExpression(ComparisonExpression {
+                        left: Box::new(Expression::Identifier(Identifier {
+                            name: "x".to_string(),
+                        })),
+                        comparison_operator: ComparisonOperator::Equal,
+                        right: Box::new(Expression::Literal(Literal::Int(42))),
+                    })),
+                    Statement::Filter(Predicate::ComparisonExpression(ComparisonExpression {
```

Comment:
Non-blocking comment: I find this sort of test painful to read because of all the syntax and at some point with code like this, I'm inclined towards adding helpers.

For example, I might suggest adding a method to make `Box::new(Expression::Identifier(Identifier(x)))` easier to read e.g., `expr_identifier(x)`. Or, `expr_comparison(left, op, right)`.